src/transformState.js

```diff
@@ -5,7 +5,31 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  if (actions.length === 0) {
+    return state;
+  }
+
+  for (const key of actions) {
+    switch (key.type) {
+      case 'addProperties' :
+        Object.assign(state, key.extraData);
+        break;
+
+      case 'removeProperties' :
+        for (const item of key.keysToRemove) {
+          delete state[item];
+        }
+        break;
+
+      case 'clear' :
+        for (const del in state) {
+          delete state[del];
+        }
+        break;
```

Comment:
Don't forget about default case for switch / case statement
You can add just simple break in it